### PR TITLE
Add rxJava module

### PR DESCRIPTION
**Pull request by ghostbuster91**

### CODE REVIEW

File: rxjava/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<manifest package="com.elpassion.android.rxjava">
+    <application/>
```
Comment (mareklangiewicz) [inline rxjava/src/main/AndroidManifest.xml]: why application tag?

File: rxjava/src/main/java/com/elpassion/android/commons/rxjava/ScheduledRxJavaCallAdapterFactory.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                                        private val subscribeOn: Scheduler) : CallAdapter.Factory() {
+
+    private val rxJavaCallAdapter by lazy { RxJavaCallAdapterFactory.createWithScheduler(subscribeOn) }
+
+    override fun get(returnType: Type, annotations: Array<out Annotation>, retrofit: Retrofit): CallAdapter<*>? {
+        if (getRawType(returnType) != Observable::class.java) {
+            return null
+        }
+        @Suppress("UNCHECKED_CAST")
+        val delegate = rxJavaCallAdapter.get(returnType, annotations, retrofit) as CallAdapter<Observable<*>>
```
Comment (mareklangiewicz) [inline rxjava/src/main/java/com/elpassion/android/commons/rxjava/ScheduledRxJavaCallAdapterFactory.kt]: Why hardcode base call adapter to RxJavaCallAdapterFactory implementation?
This way you forbid injecting any other call adapter wrappers in chain.
If you used `delegate = retrofit.nextCallAdapter...` then we could for example
implement some debug logging call adapter wrapper and inject it wherever we want.

**Review by mareklangiewicz — APPROVED** [review]
Nice to have it in 'toolbox' but I would use it sparingly, because it only saves as from writing short:
`.observeOn(...).subscribeOn(...)`
And at the same time it makes resulting observable stream less universal - for example if we want to make the real calls synchronously in some integration/manual tests.

File: rxjava/src/main/java/com/elpassion/android/commons/rxjava/ScheduledRxJavaCallAdapterFactory.kt (hunk 2 above)
Comment (ghostbuster91) [reply rxjava/src/main/java/com/elpassion/android/commons/rxjava/ScheduledRxJavaCallAdapterFactory.kt]: Yes but if anyone will not pass the rxJavaCallAdapter this adapter will crash, won't it?

**Comment by ghostbuster91** [comment]
@langara I agree, but we do have apps that are using only std async calls and this module is designed for them.

**Review by marcinsus — COMMENTED** [review]
I am skeptical about this change, in some of case it can save one additional line of code but it is very easy to make a mistake and hit application performance.

**Review by OrdonTeam — APPROVED** [review]
Looks good to me. At least as a nice reminder that this is possible. If someone dislike this idea, we don't force him to use it anyway.

File: rxjava/src/main/java/com/elpassion/android/commons/rxjava/RxUtils.kt [hunk 3]
```diff
@@ -12,3 +12,4 @@ fun <T> Observable<T>.catchOnError(handleOnError: (throwable: Throwable) -> Unit
             Observable.empty()
         }
 
+fun <T> Observable<T>.changes(init: T) = scan(init to init) { acc, cur -> acc.second to cur }.skip(1)
```
Comment (marcinsus) [inline rxjava/src/main/java/com/elpassion/android/commons/rxjava/RxUtils.kt:15]: Rename "init" param to defaultValue/initialValue

**Review by marcinsus — COMMENTED** [review]
👍

**Review by OrdonTeam — APPROVED** [review]

**Pull request merged**